SMLHelper/Options/ConfigModOptions.cs

```diff
@@ -0,0 +1,700 @@
+                {   // Method not found, log error and skip.
+                    Logger.Error($"[ConfigModOptions] Could not find the specified OnChange method: " +
+                        $"{onChangeAttribute.MethodName} in class {Config.GetType()}");
+                    continue;
+                }
+
+                var parameters = method.GetParameters();
+                var invokeParams = new object[parameters.Length];
+                var senderFound = false;
+                var eventArgsFound = false;
+                var modOptionEventFound = false;
+                for (var i = 0; i < parameters.Length; i++)
+                {
+                    var param = parameters[i];
+                    if (!senderFound && param.ParameterType == objectType)
+                    {
+                        senderFound = true;
+                        invokeParams[i] = sender;
+                        continue;
+                    }
+
+                    if (!eventArgsFound && param.ParameterType == eventArgsType)
+                    {
+                        eventArgsFound = true;
+                        invokeParams[i] = e;
+                    }
+
+                    if (!modOptionEventFound && param.ParameterType == genericModOptionEventArgsType)
+                    {
+                        modOptionEventFound = true;
+                        invokeParams[i] = e;
+                    }
+
+                    if (senderFound && eventArgsFound && modOptionEventFound)
+                        break;
+                }
+
+                method.Invoke(Config, invokeParams);
+            }
+        }
+
+        /// <summary>
+        /// Generates tooltips for each <see cref="ModOption"/> with a specified <see cref="TooltipAttribute"/>, before
+        /// invoking any relevant method(s) specified with <see cref="OnGameObjectCreatedAttribute"/>(s) and passes
+        /// parameters when a <see cref="GameObject"/> is created in the options menu.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void ConfigModOptions_GameObjectCreated(object sender, GameObjectCreatedEventArgs e)
+        {
+            if (modOptionsMetadata.TryGetValue(e.Id, out var modOptionMetadata))
+            {
+                // Create a tooltip if there is a TooltipAttribute specified
+                if (modOptionMetadata.TooltipAttribute != null)
+                {
+                    e.GameObject.GetComponentInChildren<Text>().gameObject.AddComponent<ModOptionTooltip>().Tooltip
+                        = modOptionMetadata.TooltipAttribute.Tooltip;
+                }
+
+                if (modOptionMetadata.OnGameObjectCreatedAttributes == null)
+                    return; // Skip attempting to invoke events if there are no OnGameObjectCreatedAttributes set for the member.
+
+                // cache types used in comparisons
+                var objectType = typeof(object);
```

Comment:
Same here, you don't need to create a variable to store the value of a `typeof` expression.  
It's not a method.

Reply:
https://github.com/SubnauticaModding/SMLHelper/pull/173/commits/88fbdffba0748285567429b4fdb7d1981ad5d731